src/ValueParsers/DecimalParser.php

```diff
@@ -83,14 +128,7 @@ protected function stringParse( $value ) {
 		$value = $this->unlocalizer->unlocalizeNumber( $value );
 
 		//handle scientific notation
-		if ( preg_match( '/^(.*)([eE]|x10\^)([-+]?[,\d]+)$/', $value, $matches ) ) {
-			$exponent = $this->normalizeDecimal( $matches[3] );
-			$exponent = intval( $exponent );
-
-			$value = $matches[1];
-		} else {
-			$exponent = 0;
-		}
+		list( $value, $exponent ) = $this->splitDecimalExponent( $value );
 
 		$value = $this->normalizeDecimal( $value );
```

Comment:
The exponent is now normalized inside of the private (well, should be private but is public for testing purpose) function, why is the value not normalized the same time?